### DISCUSSION TITLE
Support local presets for local platform

### DISCUSSION THREAD

**Discussion by fgreinacher** [renovatebot/renovate#22650] [Suggest an Idea]
### Tell us more.

The limitation that local presets are not working is a bit cumbersome because you have to touch the repo config to fix it (remove preset reference, copy necessary configuration from used presets).

I assume this is currently not working because renovate does not know platform/endpoint/token for local runs. Am I right with this? Could we add options to specify this information via config, e.g. `localPresetPlatfom`?

**Comment by rarkins** [comment]
This would be surprisingly hard to do, at least with the current approach for local platform.

In theory Renovate could look in the git remote and do a fairly reliable guess about the remote endpoint, and a sometimes reliable guess about the platform type. It would also need credentials.

This would seem to work better if we retained the platform/endpoint as they were, but then specified that dryRun=local or something like that

**Reply by fgreinacher** [reply]
I like the current approach with the `local` platform, because it's very nicely isolated. I think everything would be much more complicated if we implement this via `dryRun`. We would also lose the possibility to pick a dry run mode for the local run.

If we focus on enabling the preset use case, wouldn't we "just" (I know, might be a big "just" 😄) have to select another resolver at https://github.com/renovatebot/renovate/blob/82ee689f710a8ce5c9366a4e3554dc5011123f30/lib/config/presets/local/index.ts#L27?

**Reply by rarkins** [reply]
Maybe, plus credentials
